File: backend/src/main/scala/merchant/api/UpdateMenuItemCategoryApi.scala
package merchant.api

import domain.shared.given

import cats.effect.IO
import domain.merchant.UpdateMenuItemCategoryRequest
import domain.shared.{DeliveryAppState, MenuItemId, StoreId, UserRole}
import merchant.app.*
import org.http4s.HttpRoutes
import org.http4s.circe.CirceEntityCodec.*
import org.http4s.dsl.io.*
import shared.api.routing.*
import shared.app.*

val updateMenuItemCategoryApi: FixedMethodApi2[StoreId, MenuItemId, DeliveryAppState] =
  jsonPostApi2[StoreId, MenuItemId, UpdateMenuItemCategoryRequest, DeliveryAppState](
    List(routeSegment("api"), routeSegment("delivery"), routeSegment("stores")),
    List(routeSegment("menu")),
    List(routeSegment("category")),
  )

File: frontend/src/apis/merchant/UpdateMenuItemCategoryApi.ts
import type { DeliveryAppState, MenuItemId, StoreId, UpdateMenuItemCategoryRequest } from '@/objects/core/SharedObjects'
import { postNormalizedDeliveryState } from '@/system/api/DeliveryStateClient'
import { buildApiPath2, defineJsonPostApi2, routeSegment } from '@/system/api/TypedApiDefinitions'

export const updateMenuItemCategoryApiDefinition = defineJsonPostApi2<StoreId, MenuItemId, UpdateMenuItemCategoryRequest, DeliveryAppState>(
    [routeSegment('api'), routeSegment('delivery'), routeSegment('stores')],
    [routeSegment('menu')],
    [routeSegment('category')],
  )

export function updateMenuItemCategory(
  storeId: StoreId,
  menuItemId: MenuItemId,
  payload: UpdateMenuItemCategoryRequest,
) {
  return postNormalizedDeliveryState(
    buildApiPath2(updateMenuItemCategoryApiDefinition, storeId, menuItemId),
    payload,
  )
}
